hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngineImpl.java

```diff
@@ -426,10 +426,10 @@ public void reset() {
     public void shutdown(final boolean terminate) {
         logger.finest("Shutting down services...");
         waitNotifyService.shutdown();
+        operationService.shutdown();
```

Comment:
If we separate shutdown of `InvocationRegistry` and `OperationService`, I think both problems will be solved. 

See commit: https://github.com/mdogan/hazelcast/commit/1ef4f4081604cc566de435e31a50c69a65ffeb49